src/hyperlight_host/src/mem/shared_mem.rs

```diff
@@ -624,6 +665,40 @@ impl ExclusiveSharedMemory {
         Ok(self.base_addr() + offset)
     }
 
+    /// Fill the memory in the range `[offset, offset + len)` with `value`
+    #[instrument(err(Debug), skip_all, parent = Span::current(), level= "Trace")]
+    pub fn zero_fill(&mut self, offset: usize, len: usize) -> Result<()> {
+        bounds_check!(offset, len, self.mem_size());
+        let data = self.as_mut_slice();
+        data[offset..offset + len].fill(0);
+        self.mark_pages_dirty(offset, len)?;
+        Ok(())
+    }
+
+    /// Same as `copy_from_slice` but doesn't dirty the pages.
+    /// # Safety
+    /// This function is unsafe because it does not mark the pages as dirty.
+    /// Only use this if you are certain that the pages do not need to be marked as dirty.
+    #[instrument(err(Debug), skip_all, parent = Span::current(), level= "Trace")]
```

Comment:
What circumstances is this used under?

Reply:
We need these two for restoring a snapshot. Restoring a sandbox to a snapshotted state should not itself dirty the pages